clang-tidy-checks/max-tokens-check.diff

```diff
@@ -0,0 +1,285 @@
+diff --git a/clang-tools-extra/clang-tidy/misc/CMakeLists.txt b/clang-tools-extra/clang-tidy/misc/CMakeLists.txt
```

Comment:
since `setup.sh` isn't in this PR I can't review it explicitly, but I ran into an issue when building (cmake doesn't like the `build` folder since it's already in the `build` folder from the earlier `cd build`, so it ends up looking for `.../llvm-project/build/build`)

```diff
diff a/clang-tidy-checks/setup.sh b/clang-tidy-checks/setup.sh
index 501d92f..d8724fa 100755
--- a/clang-tidy-checks/setup.sh
+++ b/clang-tidy-checks/setup.sh
@@ -47,7 +47,7 @@ function build() {
         -DLLVM_BUILD_TOOLS=OFF \
         -DLLVM_BUILD_UTILS=OFF \
         -GNinja ../llvm
-  cmake --build build
+  cmake --build .
 }
 
 function setup() {
```

and while you're in there could you add this error message as well so re-runs of `setup.sh` tell you what to do when it goes wrong:

```diff
@@ -22,7 +22,7 @@ function check_requirements() {
 
 function clone_llvm() {
   info "cloning llvm"
-  git clone -b llvmorg-11.0.0 https://github.com/llvm/llvm-project.git --depth=1
+  git clone -b llvmorg-11.0.0 https://github.com/llvm/llvm-project.git --depth=1 || (echo "checkout already exists, delete it with 'rm -r llvm-project' and try again" && exit 1)
   success
 }

```

Reply:
also the build command doesn't seem to actually build clang-tidy? all I get are a bunch of llvm binaries, is it because of the 

```bash
...
-GNinja ../llvm
```

Reply:
for reference I used these commands to get the build working (probably is doing some extra work thats not necessary though):

```bash
cd llvm-project
cmake -S llvm -B build -G Ninja -DLLVM_ENABLE_PROJECTS="clang;clang-tools-extra"
cmake --build build
```

also looks like `setup.sh` doesn't patch in this diff either (it should probably just `cat ../*.diff | patch ...`

Reply:
@driazati @samestep I've resolved the setup script issues and have and tested it. It should now support patching multiple diffs, and should build the correct binaries